### PR TITLE
Explicitly setting CertManager APIVersion to V1

### PR DESCRIPTION
**Pull request by hawkesn**
### Issue
https://github.com/aws/eks-charts/issues/942
and
Fixes: #3188

### Description
This PR explicitly sets v1 for the certmanager API version:
`apiVersion: cert-manager.io/v1`

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### Testing

Ran `helm template --set clusterName=test .`
```
---
# Source: aws-load-balancer-controller/templates/webhook.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: MutatingWebhookConfiguration
metadata:
  name: aws-load-balancer-webhook
  labels:
    helm.sh/chart: aws-load-balancer-controller-1.5.3
    app.kubernetes.io/name: aws-load-balancer-controller
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v2.5.1"
    app.kubernetes.io/managed-by: Helm
webhooks:
- clientConfig:
    caBundle: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURRRENDQWlpZ0F3SUJBZ0lSQU9qaDIxMFlBakRsUjdsK2hqNUluN2d3RFFZSktvWklodmNOQVFFTEJRQXcKS..........
```
This continues to function as normal.

Ran `helm template --set clusterName=test --set enableCertManager=true .`
```
---
# Source: aws-load-balancer-controller/templates/webhook.yaml
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: aws-load-balancer-serving-cert
  namespace: kube-system
  labels:
    helm.sh/chart: aws-load-balancer-controller-1.5.3
    app.kubernetes.io/name: aws-load-balancer-controller
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v2.5.1"
    app.kubernetes.io/managed-by: Helm
spec:
  dnsNames:
  - aws-load-balancer-webhook-service.kube-system.svc
  - aws-load-balancer-webhook-service.kube-system.svc.cluster.local
  issuerRef:
    kind: Issuer
    name: aws-load-balancer-selfsigned-issuer
  secretName: aws-load-balancer-tls
---
# Source: aws-load-balancer-controller/templates/webhook.yaml
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: aws-load-balancer-selfsigned-issuer
  namespace: kube-system
  labels:
    helm.sh/chart: aws-load-balancer-controller-1.5.3
    app.kubernetes.io/name: aws-load-balancer-controller
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/version: "v2.5.1"
    app.kubernetes.io/managed-by: Helm
spec:
  selfSigned: {}
```
apiVersion of cert-manager is set to v1.

### CODE REVIEW

File: helm/aws-load-balancer-controller/templates/webhook.yaml [hunk 1]
```diff
@@ -212,11 +212,7 @@ data:
   tls.crt: {{ $tls.clientCert }}
   tls.key: {{ $tls.clientKey }}
 {{- else }}
-{{- if .Capabilities.APIVersions.Has "cert-manager.io/v1" }}
-apiVersion: cert-manager.io/v1
-{{- else }}
-apiVersion: cert-manager.io/v1alpha2
-{{- end }}
+apiVersion: cert-manager.io/{{ .Values.certManagerApiVersion }}
```
Comment (kishorj) [inline helm/aws-load-balancer-controller/templates/webhook.yaml:215]: Lets specify the v1 version explicitly and we specify the min cert-manager version in our documents.

**Comment by johngmyers** [comment]
Why would one want to explicitly specify this?

**Comment by kishorj** [comment]
> Why would one want to explicitly specify this?

Since `cert-manager.io/v1` is the stable version, we should stick with it rather than providing an option to choose between v1alpha2 and v1 unless there is a compelling reason. This will also simplify future maintenance.

**Comment by johngmyers** [comment]
My question was to the author as to why an admin would want to explicitly specify apiVersion. The existing `.Capabilities.APIVersions.Has` mechanism is reasonable for a transitition time.

The v1 API was introduced in cert-manager 1.0.0, released September 2020. It's probably time to drop support for older cert-manager.

**Comment by hawkesn** [comment]
> My question was to the author as to why an admin would want to explicitly specify apiVersion. The existing `.Capabilities.APIVersions.Has` mechanism is reasonable for a transitition time.
> 
> The v1 API was introduced in cert-manager 1.0.0, released September 2020. It's probably time to drop support for older cert-manager.

I agree with you that we should drop support for the older cert-manager, I let a user specify it for backwards compatibility reasons.

File: helm/aws-load-balancer-controller/README.md [hunk 2]
```diff
@@ -82,6 +82,8 @@ kubectl apply -k "github.com/aws/eks-charts/stable/aws-load-balancer-controller/
 
 If you are setting `enableCertManager: true` you need to have installed cert-manager and it's CRDs before installing this chart; to install [cert-manager](https://artifacthub.io/packages/helm/cert-manager/cert-manager) follow the installation guide.
 
+By default, the apiVersion of cert-manager is set to `cert-manager.io/v1`.
```
Comment (kishorj) [inline helm/aws-load-balancer-controller/README.md:85]:
```suggestion
The controller helm chart requires the cert-manager with apiVersion `cert-manager.io/v1`.
```

**Review by kishorj — APPROVED** [review]

**Comment by kishorj** [comment]
@sudermanjr, we will need a new chart release. Chart version 1.5.3 doesn't contain the changes from this PR.

**Pull request merged**